core/tests/flow.rs

```diff
@@ -2,336 +2,219 @@
+use clementine_core::musig2::aggregate_nonces;
+use clementine_core::servers::*;
+use clementine_core::traits::rpc::OperatorRpcClient;
+use clementine_core::traits::rpc::VerifierRpcClient;
+use clementine_core::user::User;
+use clementine_core::EVMAddress;
+use clementine_core::UTXO;
+use clementine_core::{
+    create_extended_rpc, create_test_config, create_test_config_with_thread_name,
+};
+use std::thread;
+
+#[tokio::test]
+async fn test_deposit() -> Result<(), BridgeError> {
+    let mut config = create_test_config_with_thread_name!("test_config_flow_1.toml");
+    let rpc = create_extended_rpc!(config);
+
+    // Create temporary databases for testing.
+    let handle = thread::current()
+        .name()
+        .unwrap()
+        .split(':')
+        .last()
+        .unwrap()
+        .to_owned();
+    for i in 0..4 {
+        create_test_config!(
+            handle.clone() + i.to_string().as_str(),
+            "test_config_flow_1.toml"
+        );
+    }
+
+    let mut operators = Vec::new();
+    let mut evm_addresses = Vec::new();
+    for i in 0..config.num_operators {
+        let secret_key = config.all_operators_secret_keys.clone().unwrap()[i].clone();
+        let operator = create_operator_server(
+            BridgeConfig {
+                secret_key,
+                ..config.clone()
+            },
+            rpc.clone(),
+        )
+        .await
+        .unwrap();
+
+        operators.push((operator.0, operator.1, secret_key));
+        evm_addresses.push(EVMAddress([(i + 1) as u8; 20]));
+    }
+    let mut verifiers = Vec::new();
+    for i in 0..config.num_verifiers {
+        verifiers.push(
+            create_verifier_server(
+                BridgeConfig {
+                    secret_key: config.all_verifiers_secret_keys.clone().unwrap()[i].clone(),
+                    ..config.clone()
+                },
+                rpc.clone(),
+            )
+            .await
+            .unwrap(),
+        );
+    }
+
+    println!("Operators: {:#?}", operators);
+    println!("Verifiers: {:#?}", verifiers);
+
+    let secret_key = secp256k1::SecretKey::new(&mut secp256k1::rand::thread_rng());
+
+    let user = User::new(
+        rpc.clone(),
+        config.verifiers_public_keys.clone(),
+        secret_key,
+        config.clone(),
+    );
+
+    let deposit_outpoints: Vec<_> = evm_addresses
+        .iter()
+        .map(|&address| {
+            let deposit_address = user.get_deposit_address(address).unwrap();
+            let deposit_outpoint = rpc
+                .send_to_address(&deposit_address, BRIDGE_AMOUNT_SATS)
+                .unwrap();
+
+            deposit_outpoint
+        })
+        .collect();
+
+    rpc.mine_blocks(18).unwrap();
+
+    // for every verifier, we call new_deposit
+    // aggregate nonces
+    let mut agg_nonces = Vec::new();
+
+    for (i, (client, _)) in verifiers.iter().enumerate() {
+        let musig_pub_nonces = client
+            .verifier_new_deposit_rpc(
```

Comment:
All the verifiers should get the same deposit request, so that they will sign a correct thing